lib/openstudio/alfalfa/creator.rb

```diff
@@ -150,13 +153,111 @@ def resolve_template(mapping)
+        entity_info['relationships'] = {} unless entity_info['relationships']
+        entity_info['relationships'][relationship[@metadata_type.downcase]] = OpenStudio.removeBraces(parent_obj.handle)
+        return entity_info
+      end
+
+      ##
+      # Add nodes defined in mapping document as entities
+      ##
+      # @param [OpenStudio parent object] obj
+      # @param [Hash] nodes
+      def add_nodes(obj, nodes)
+        relationship_to_parent = nodes['relationship_to_parent']
+        nodes.each do |node_method, node_properties|
+          next unless node_method != 'relationship_to_parent'
+          found_node = obj.send(node_method)
+          found_node = found_node.get unless found_node.is_a?(OpenStudio::Model::Node)
+          next unless found_node.initialized
+          node_properties.each do |system_node_property, map|
+            name = "#{obj.name} #{map['brick']}" # Brick names are prettier / consistent
+            name = create_ems_str(name)
+
+            # Else recreates variable every time
+            output_variable = @model.getOutputVariableByName(name)
+            if output_variable.is_initialized
+              output_variable = output_variable.get
+            else
+              output_variable = create_output_variable_and_ems_sensor(system_node_property: system_node_property, node: found_node, ems_name: name, model: @model)
+            end
+            entity_info = resolve_template(map[@metadata_type.downcase])
+            entity_info = add_node_relationship_to_parent(obj, relationship_to_parent, entity_info) unless relationship_to_parent.nil?
+            add_specific_info(output_variable, entity_info)
+          end
+        end
+      end
+
+      # Necessary when adding additional output / EMS variables
+      # so they get stored in OSM
+      def save_model
+        @model.save(@path_to_model, true)
+      end
+
+      ##
+      # @return [Boolean or One of AirLoopHVACUnitary* objects]
+      def check_if_component_is_unitary(sc)
+        r = false
+        if sc.to_AirLoopHVACUnitaryHeatPumpAirToAir.is_initialized
+          r = sc.to_AirLoopHVACUnitaryHeatPumpAirToAir
+        elsif sc.to_AirLoopHVACUnitarySystem.is_initialized
+          r = sc.to_AirLoopHVACUnitarySystem
+        elsif sc.to_AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed.is_initialized
+          r = sc.to_AirLoopHVACUnitaryHeatPumpAirToAirMultiSpeed
+        end
+        if r
+          r = r.get
+        end
+        return r
+      end
+
+      ##
+      # Deletes all of the AirLoopHVAC entities in favor of the contained unitary system
+      # and replaces the unitary entity id with the airloop id.
+      def resolve_unitary_and_air_loops_overlap
+        handles_to_swap = {}
+        air_loops = @model.getAirLoopHVACs
+        air_loops.each do |air_loop|
+          supply_components = air_loop.supplyComponents
+          supply_components.each do |sc|
```

Comment:
you could probably get rid of the `air_loops` and `supply_components` variables as well, it doesn't look like they're used anywhere outside of as an iterator